Python/Module2_EssentialsOfPython/Basic_Objects.md

```diff
@@ -721,11 +721,77 @@ True
 
 ```
 
+### Formatting strings
+Python provides multiple syntaxes for formatting strings; these permit us to do things like programmatically inject the values of variables into strings, align fields using whitespace, and control the number of decimal places with which numbers are displayed in a string. This section is designed to simply expose the reader to the different varieties of string-formatting. 
+
+[pyformat.info](https://pyformat.info) is the best resource to consult to see an exhaustive (but still intuitive) treatment of string-formatting in Python. You can also refer to the official documentation [here](https://docs.python.org/3/library/string.html#format-examples).
+
+In Python 3, you can leverage the `format` method towards this end:
+
+```python
+# using `format` to replace placeholders with values
+>>> name = "Bruce"
+>>> age = 80
+>>> "{name} is {age} years old".format(name="Bruce", age=80)
+'Bruce is 80 years old'
+
+# padding a string with 4 leading spaces
+>>> "{item:>4}".format(item="stew")
+'   stew'
```

Comment:
This is not correct.

```python
In [1]: '{item:>4}'.format(item='stew')                                                        
Out[1]: 'stew'
```

You probably want `item:>8`

Reply:
Whew! Good catch!